### PR TITLE
Submission Contact Fragment & StepView Custom View

### PR DESCRIPTION
**Pull request by Fabian-K**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes #41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unncessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Check our [Contribution Guidelines](https://github.com/corona-warn-app/cwa-app-android/blob/master/CONTRIBUTING.md)

## Description
This PR adds the Submission Contact Fragment which allows to call for a teletan. 

image

It also includes a custom view StepEntry that can be reused to display fragments like the following: 
image
The use of a custom view compared to `include_test_result_step.xml` is more flexible with respect to the body and enables composability. Also it renders correctly in the layout preview. If accepted, I plan to replace include_test_result_step with SimpleStepEntry in a following PR.

Looking forward to your feedback!

Thanks, Fabian

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionContactFragment.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        binding.submissionContactButtonCall.setOnClickListener {
+            dial()
+        }
+        binding.submissionContactButtonEnter.setOnClickListener {
+            doNavigate(SubmissionContactFragmentDirections.actionSubmissionContactFragmentToSubmissionTanFragment())
+        }
+    }
+
+    private fun dial() = context?.let {
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionContactFragment.kt:48]: we already have a CallHelper in place, please use this one.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    val body: FrameLayout?
+
+    init {
+        inflate(context, R.layout.view_step_entry, this)
+
+        body = findViewById(R.id.step_entry_framelayout_body)
+
+        context.withStyledAttributes(attrs, R.styleable.StepEntry) {
+            val icon = getDrawable(R.styleable.StepEntry_step_entry_icon)
+            findViewById<ImageView>(R.id.step_entry_imageview_icon).setImageDrawable(icon)
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt:30]: please use `kotlinx.android.synthetic...` to access the view

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+            val isFinal = getBoolean(R.styleable.StepEntry_step_entry_final, false)
+            findViewById<View>(R.id.step_entry_view_line).visibility = if (isFinal) {
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/StepEntry.kt:33]: please use `kotlinx.android.synthetic...` to access the view

File: Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/simple_step_entry_textview_title"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml:9]: we don't use textview in the id, the type of the element should be clear through `_title`

File: Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        style="@style/textTitle"
+        android:layout_width="@dimen/match_constraint"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Title" />
+
+    <TextView
+        android:id="@+id/simple_step_entry_textview_body"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_step_entry_simple_body.xml:19]: we don't use textview in the id, the type of the element should be clear through `_body`

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        style="@style/fadingScrollView"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+
+            <ImageView
+                android:id="@+id/submission_contact_imageview_illustration"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:15]: we don't use imageview in the id, the type of the element should be clear through `_illustration`

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+                android:layout_width="@dimen/match_constraint"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_information_illustration_contact"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/submission_contact_textview_body"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:24]: we don't use textview in the id, the type of the element should be clear through `_body`

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+                android:layout_width="@dimen/match_constraint"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/spacing_small"
+                android:text="@string/submission_contact_text"
+                app:layout_constraintEnd_toStartOf="@+id/guideline_end"
+                app:layout_constraintStart_toStartOf="@+id/guideline_start"
+                app:layout_constraintTop_toBottomOf="@id/submission_contact_imageview_illustration" />
+
+            <TextView
+                android:id="@+id/submission_contact_textview_headline"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:35]: we don't use textview in the id, the type of the element should be clear through `_headline`

File: Corona-Warn-App/src/main/res/layout/view_step_entry.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            android:id="@+id/step_entry_imageview_icon"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:background="@drawable/ic_step_background"
+            android:src="@drawable/ic_step_1"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <FrameLayout
+            android:id="@+id/step_entry_framelayout_body"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/view_step_entry.xml:19]: looks like an unused FrameLayout

File: Corona-Warn-App/src/main/res/layout/view_step_entry.xml [hunk 10]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <ImageView
+            android:id="@+id/step_entry_imageview_icon"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/view_step_entry.xml:10]: remove imageview from id

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+                        android:layout_width="@dimen/match_constraint"
+                        android:layout_height="wrap_content"
+                        android:layout_marginTop="@dimen/button_icon_margin"
+                        android:text="@string/submission_contact_step_1_body"
+                        app:layout_constraintEnd_toEndOf="parent"
+                        app:layout_constraintStart_toStartOf="parent"
+                        app:layout_constraintTop_toTopOf="parent" />
+
+                    <TextView
+                        android:id="@+id/submission_contact_textview_step_1_number"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:72]: remove textview

File: Corona-Warn-App/src/main/res/layout/include_submission_contact.xml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+                app:layout_constraintTop_toBottomOf="@id/submission_contact_textview_headline"
+                app:step_entry_final="false"
+                app:step_entry_icon="@drawable/ic_step_1">
+
+                <androidx.constraintlayout.widget.ConstraintLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content">
+
+                    <TextView
+                        android:id="@+id/submission_contact_textview_step_1_body"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/include_submission_contact.xml:62]: we don't use textview in the id, the type of the element should be clear through `_title`

**Review by harambasicluka — CHANGES_REQUESTED** [review]
The ID misunderstandings are due to a non-unique [naming convention](https://github.com/corona-warn-app/cwa-app-android/wiki/3-UI#view-ids).

File: Corona-Warn-App/src/main/res/layout/view_step_entry.xml (hunk 9 above)
Comment (Fabian-K) [reply Corona-Warn-App/src/main/res/layout/view_step_entry.xml:19]: This is the insertion point for the children of the StepEntry:

```
<de.rki.coronawarnapp.ui.view.StepEntry>
  <TextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="@string/submission_contact_step_2_body" />
</de.rki.coronawarnapp.ui.view.StepEntry>
```

**Comment by Fabian-K** [comment]
🙏 for the feedback! I applied the changes and added one comment regarding the FrameLayout.

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**